### PR TITLE
fix: Fix get_header_view panic bug

### PR DESCRIPTION
**Pull request by TheWaWaR**
Update best headers(peers/global) after update header_map

### CODE REVIEW

File: sync/src/types.rs [hunk 1]
```diff
@@ -727,6 +727,7 @@ impl SyncState {
     }
 
     pub fn set_shared_best_header(&self, header: HeaderView) {
+        assert!(self.header_map.read().contains_key(&header.hash()));
```
Comment (doitian) [inline sync/src/types.rs]: Give an assertion message.

**Comment by quake** [comment]
bors r=quake,u2,keroro520,driftluo

**Comment by TheWaWaR** [comment]
bors r=quake,u2,keroro520,driftluo

**Comment by zhangsoledad** [comment]
@doitian

**Comment by doitian** [comment]
bors r=quake,u2,keroro520,driftluo

**Pull request merged**